src/result_processor.c

```diff
@@ -898,14 +894,11 @@ static int rpSafeLoaderNext_Accumulate(ResultProcessor *rp, SearchResult *res) {
   // Done loading. Unlock Redis
   RedisModule_ThreadSafeContextUnlock(sctx->redisCtx);
 
-  if (isQueryProfile) {
+  if (isQueryProfile){
     clock_gettime(CLOCK_MONOTONIC, &rpEndTime);
-  } else {
-    rpEndTime.tv_nsec = 0;
+    rs_timersub(&rpEndTime, &rpStartTime, &rp->GILTime);
```

Comment:
In some cases, we may use the `LOADER` multiple times. We should sum up the GIL time here as well, and not override it